src/python/gem5/components/boards/kernel_disk_workload.py

```diff
@@ -97,7 +95,11 @@ def get_disk_device(self) -> str:
         raise NotImplementedError
 
     @abstractmethod
-    def _add_disk_to_board(self, disk_image: DiskImageResource) -> None:
```

Comment:
Let's do this a bit different. Here's my worry: By doing this we're forcing change in all boards that inherit from `KernelDiskWorkload` to support multiple disks. This may not always be wanted and certainly won't be the case when this is first implemented.

How about:

```py

def _add_disk_to_board(self, disk_image: Disk ImageResource) -> None:
    """
    Sets the configuration needed to add the disk image to the board.
    .. note::
        This will be executed at the end of the
        ``set_kernel_disk_workload`` function.
    :param disk_image: The disk image to add to the system.
    """
    raise NotImplementedError

def _supports_multiple_images(cls) -> bool:
   """Returns True if the board supports adding multiple additional
   disk images beyond just the root disk image.
   """
   return False

def _add_additional_disks_to_board(
    self,
    additional_disk_images: List[DiskImageResource],
) -> None:
    """<TODO SOME DOCSTRING HERE>"""
    assert self. _supports_multiple_images(), \
        "`_add_additional_disks_to_board` called when `_supports_multiple_images` returned0 False."
    raise NotImplementedError


# Then, down in the `set_kernel_disk_workload` function:

self._add_disk_to_board(disk_image=disk_image)
if self._supports_multiple_images():
    self._add_additional_disk_images_to_board(additional_images = additional_images)
elif len(additional_images) > 0:
    warn("This board does not support multiple disk images. Additional images will not be added to the board.")
```

With this, for boards that support multiple disk images, you'd just need to override the `_supports_multiple_disk_images` function to return `True`. Nothing would need to change in any boards we don't need or want to implement this feature yet.

Reply:
I think your idea is very interesting, I've already applied it.